src/modify.js

```diff
@@ -80,10 +94,21 @@ function rewriteAllFields(schema, configCallback, context) {
 
 export function modify(originalSchema, config) {
   const schema = JSON.parse(JSON.stringify(originalSchema));
+  // All these functions mutate "schema" that's why we create a copy above
 
-  rewriteFields(schema, config.fields);
-
+  const resultRewrite = rewriteFields(schema, config.fields);
   rewriteAllFields(schema, config.allFields);
 
-  return schema;
+  if (!config.muteWarningTip) {
+    console.warn(
+      'json-schema-form modify(): We highly recommend you to handle/report the returned `warnings` as they highlight possible bugs in your modifications. To mute this log, pass `muteWarningTip: true` to the config.'
+    );
+  }
+
+  const allWarnings = [resultRewrite?.warnings].filter(Boolean).flat();
```

Comment:
You need to flatten before the filter. Also no need for the null check since `resultRewrite` always returns an object

Reply:
@danielcardoso5 nice catch on the flat to be before, but the `warnings` can be null if the config does not exist (line 31). But I changed the line to return it. Check [b68152d](https://github.com/remoteoss/json-schema-form/pull/83/commits/b68152df3be6f53106d9058f24a90bed994e26c3)